libdrizzle/field.cc

```diff
@@ -402,9 +402,17 @@ drizzle_return_t drizzle_state_binary_field_read(drizzle_st *con)
 {
   drizzle_return_t ret;
 
-  switch(con->result->column_buffer[con->result->field_current].type)
+  /* Skip NULL fields */
+  while (con->result->null_bitmap[(con->result->field_current_read + 2) / 8] & (1 << ((con->result->field_current_read + 2) % 8)))
```

Comment:
1. Yes, and I've captured the actual packet on wire using WireShark, confirmed that 2-bits offset is actually there, but in fact I have no idea why, and I found nothing about this magic number in that manual either :(  Maybe they want to maintain protocol compatibility ?

2. OK, I'll do it tomorrow :)